feat(oxfmt): Support TOML(v1.0 only) files

leaysgur opened the pull request:
Fixes #16907, closes #17048 

- Basically, the same approach with `prettier-plugin-toml`
  - https://github.com/ikatyang-collab/linguist-languages/blob/main/data/TOML.js
  - https://github.com/un-ts/prettier/blob/master/packages/toml/src/index.ts#L27-L31

I will add tests in next PR.
Also, I will add improvement to reduce cloning options.

leaysgur commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/oxc-project/oxc/17113?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#17113** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/17113?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/17113?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#17109** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/17109?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=17113).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen reviewed (commented):
Feel free to merge, I'll clone the repo to remove all the unwanted dependencies.

leaysgur committed:
feat(oxfmt): Support TOML(v1.0 only) files (#17113)

Fixes #16907, closes #17048

- Basically, the same approach with `prettier-plugin-toml`
  - https://github.com/ikatyang-collab/linguist-languages/blob/main/data/TOML.js
  - https://github.com/un-ts/prettier/blob/master/packages/toml/src/index.ts#L27-L31

I will add tests in next PR.
Also, I will add improvement to reduce cloning options.
